aws_lambda_powertools/utilities/data_classes/common.py

```diff
@@ -35,6 +35,10 @@ def get_header_value(
     if case_sensitive:
         return headers.get(name, default_value)
 
+    # If headers is NoneType, return default value
+    if not headers:
+        return default_value
+
```

Comment:
```suggestion
    # If headers is NoneType, return default value
    if not headers:
        return default_value

    if case_sensitive:
        return headers.get(name, default_value)
```

Changing the order to prevent `case_sensitive=True` logic to kick in and fail on `AttributeError`, since `None` wouldn't have `.get`